cosmpy/aerial/tx.py

```diff
@@ -41,6 +42,163 @@
+        payer: Optional[Address] = None,
+    ) -> "TxFee":
+        """Init the Transaction fee class from string amount.
+
+        :param amount: tx fee amount string
+        :param gas_limit: gas limit
+        :param granter: transaction fee granter, defaults to None
+        :param payer: transaction fee payer, defaults to None
+        :return: TxFee
+        """
+        return cls(
+            amount=parse_coins(amount),
+            gas_limit=gas_limit,
+            granter=granter,
+            payer=payer,
+        )
+
+    @classmethod
+    def from_gas_only(
+        cls,
+        client: "LedgerClient",  # type: ignore # noqa: F821
+        gas_limit: int,
+        granter: Optional[Address] = None,
+        payer: Optional[Address] = None,
+    ) -> "TxFee":
+        """Init the Transaction fee class from gas limit.
+
+        :param client: Ledger client
+        :param gas_limit: gas limit
+        :param granter: transaction fee granter, defaults to None
+        :param payer: transaction fee payer, defaults to None
+        :return: TxFee
+        """
+        estimated_amount = client.estimate_fee_from_gas(gas_limit)
+        return cls(
+            amount=parse_coins(estimated_amount),
+            gas_limit=gas_limit,
+            granter=granter,
+            payer=payer,
+        )
+
+    @classmethod
+    def from_simulation(
+        cls,
+        client: "LedgerClient",  # type: ignore # noqa: F821
+        tx: "Transaction",  # type: ignore  # noqa: F821
+        sender: "Wallet",  # type: ignore # noqa: F821
+        amount: Optional[str] = None,
+        granter: Optional[Address] = None,
+        payer: Optional[Address] = None,
+        account: Optional["Account"] = None,  # type: ignore # noqa: F821
+        memo: Optional[str] = None,
+    ) -> Tuple["TxFee", Optional["Account"]]:  # type: ignore # noqa: F821
+        """Estimate transaction fees based on either a provided amount, gas limit, or simulation.
+
+        :param client: Ledger client
+        :param tx: The transaction
+        :param sender: The transaction sender
+        :param amount: Transaction fee amount, defaults to None
+        :param granter: Transaction fee granter, defaults to None
+        :param payer: Transaction fee payer, defaults to None
+        :param account: The account
+        :param memo: Transaction memo, defaults to None
+        :return: Tuple[TxFee, Optional[Account]]
+        """
+        # Ensure we have the account info
+        account = account or client.query_account(sender.address())
+
+        # Simulate transaction to get gas and amount
+        gas_limit, estimated_amount = simulate_tx(client, tx, sender, account, memo)
+
+        # Use estimated amount if not provided
+        amount = amount or estimated_amount
+
+        fee = cls(
+            amount=parse_coins(amount),
+            gas_limit=gas_limit,
+            granter=granter,
+            payer=payer,
+        )
+
+        return fee, account
```

Comment:
I would relocate this method to the `Tx` class - it feels like it would fit there better than here.